gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java

```diff
@@ -50,6 +50,11 @@ public HttpRequest(HttpMethod method, String uri, MultiMap headers, byte[] paylo
 
     /**
      * @param object object
+     * @throws IllegalArgumentException
+     *      When passed in JSON is not in expected format.
+     * @throws ClassCastException
```

Comment:
Does it throw both exceptions now? If not change the javadoc

Reply:
There's nothing new with those exceptions. They were already there before my change. I only documented them because that was missing and would help a lot to write more reliable code. Here some links pointing to previous release:

**IllegalArgumentException**:
- Thrown at lines [58](https://github.com/swisspush/gateleen/blob/develop/gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java#L58), [67](https://github.com/swisspush/gateleen/blob/develop/gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java#L67) and [73](https://github.com/swisspush/gateleen/blob/develop/gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java#L73).

**ClassCastException**:
- Thrown at lines [55](https://github.com/swisspush/gateleen/blob/develop/gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java#L55), [56](https://github.com/swisspush/gateleen/blob/develop/gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java#L56), [69](https://github.com/swisspush/gateleen/blob/develop/gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java#L69) and [73](https://github.com/swisspush/gateleen/blob/develop/gateleen-core/src/main/java/org/swisspush/gateleen/core/http/HttpRequest.java#L73).